pkg/controller/installation/core_controller.go

```diff
@@ -1257,23 +1179,37 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	}
 	// Build a configuration for rendering calico/node.
 	nodeCfg := render.NodeConfiguration{
-		K8sServiceEp:              k8sapi.Endpoint,
-		Installation:              &instance.Spec,
-		AmazonCloudIntegration:    aci,
-		LogCollector:              logCollector,
-		BirdTemplates:             birdTemplates,
-		TLS:                       typhaNodeTLS,
-		ClusterDomain:             r.clusterDomain,
-		NodeReporterMetricsPort:   nodeReporterMetricsPort,
-		BGPLayouts:                bgpLayout,
-		NodeAppArmorProfile:       nodeAppArmorProfile,
-		MigrateNamespaces:         needNsMigration,
-		Terminating:               nodeTerminating,
-		PrometheusServerTLS:       nodePrometheusTLS,
-		PrometheusMetricsCABundle: metricsBundle,
+		K8sServiceEp:            k8sapi.Endpoint,
+		Installation:            &instance.Spec,
+		AmazonCloudIntegration:  aci,
+		LogCollector:            logCollector,
+		BirdTemplates:           birdTemplates,
+		TLS:                     typhaNodeTLS,
+		ClusterDomain:           r.clusterDomain,
+		NodeReporterMetricsPort: nodeReporterMetricsPort,
+		BGPLayouts:              bgpLayout,
+		NodeAppArmorProfile:     nodeAppArmorProfile,
+		MigrateNamespaces:       needNsMigration,
+		Terminating:             nodeTerminating,
+		PrometheusServerTLS:     nodePrometheusTLS,
 	}
 	components = append(components, render.Node(&nodeCfg))
 
+	components = append(components,
+		rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
+			Namespace:       common.CalicoNamespace,
+			ServiceAccounts: []string{render.CalicoNodeObjectName, render.TyphaServiceAccountName},
+			KeyPairOptions: []rcertificatemanagement.KeyPairOption{
+				// this controller is responsible for rendering the tigera-ca-private secret.
+				rcertificatemanagement.NewKeyPairOption(certificateManager.KeyPair(), true, false),
```

Comment:
I'm concerned there might be a regression here, though I honestly am not sure, there may be something I'm missing.
Looking at this I think it looks like the KeyPairs here are always returned by the CertificateManagement component (in the Objects function). That means that they are all given Ownership references even if they were not originally created by the operator.
I'm not sure I'm reading this right because I see there is still a UT for "should not add OwnerReference(OR) to user supplied node and typha certs". Or maybe that test still works when they are the previous cert style, but the new style would always have the OR added.
Specifically the problem is if a user provides one of these secrets, like the certificateManager.KeyPair, the operator should not be placing an OR on the resource the user created (in the operator namespace, setting an owner reference when we copy a secret to the namespace where it is needed we should add an OR).

Reply:
If the renderInOperator namespace is true, it does not yet mean that it will overwrite the secret that a user has brought to the cluster, I suppose it is accurate to conclude that this controller creates the secret, if it is absent.

When we go to certificatemanagement.go the keypair will be tested in these lines:
```
if keyPairCreator.renderInOperatorNamespace && 
(!keyPair.BYO() || keyPair.GetName() == certificatemanagement.CASecretName) {
					objsToCreate = append(objsToCreate, keyPair.Secret(common.OperatorNamespace()))
				}
```
This means that if the keyPair.BYO() is true, nothing will be rendered for this keypair. BYO() would return true if a user has created the secret.